lib/dor/workflow/client/lifecycle_routes.rb

```diff
@@ -10,33 +11,101 @@ def initialize(requestor:)
+            version = opts[:version]
+            active_only = opts.key?(:active_only) ? opts[:active_only] : false
+          else
+            raise ArgumentError, 'wrong number of arguments, must be 1-5'
+          end
+
+          Deprecation.warn(self, 'passing the repo parameter to lifecycle is no longer necessary. This will raise an error in dor-workflow-client version 4') if repo
+
+          filter_milestone(query_lifecycle(druid, version: version, active_only: active_only), milestone_name)
         end
 
         # Returns the Date for a requested milestone ONLY for the current version.
         # This is slow as the workflow server will query dor-services-app for the version.
         # A better approach is #lifecycle with the version tag.
-        # @param [String] repo repository name
         # @param [String] druid object id
         # @param [String] milestone_name the name of the milestone being queried for
         # @param [Number] version the version to query for
         # @return [Time] when the milestone was achieved.  Returns nil if the milestone does not exis
         #
-        def active_lifecycle(repo, druid, milestone_name, version: nil)
-          lifecycle(repo, druid, milestone_name, version: version, active_only: true)
+        def active_lifecycle(*args)
+          case args.size
+          when 4
+            Deprecation.warn(self, 'you provided 4 args, but active_lifecycle now takes kwargs')
+            (repo, druid, milestone_name) = args[0..2]
+            version = args[3][:version]
+          when 3
+            Deprecation.warn(self, 'you provided 3 args, but active_lifecycle now takes kwargs')
+            (repo, druid, milestone_name) = args
+            version = nil
+          when 2
+            Deprecation.warn(self, 'you provided 2 args, but active_lifecycle now takes kwargs')
+            (druid, milestone_name) = args
+            version = nil
+          when 1
+            opts = args.first
+            repo = opts[:repo]
+            druid = opts[:druid]
+            milestone_name = opts[:milestone_name]
+            version = opts[:version]
+          else
+            raise ArgumentError, 'wrong number of arguments, must be 1-4'
+          end
+
+          Deprecation.warn(self, 'passing the repo parameter to active_lifecycle is no longer necessary. This will raise an error in dor-workflow-client version 4') if repo
+
+          lifecycle(druid: druid, milestone_name: milestone_name, version: version, active_only: true)
         end
 
         # @return [Array<Hash>]
-        def milestones(repo, druid)
-          doc = query_lifecycle(repo, druid, active_only: false)
+        def milestones(*args)
+          case args.size
+          when 2
+            Deprecation.warn(self, 'you provided 2 args, but active_lifecycle now takes kwargs')
+            (repo, druid) = args
+          when 1
+            opts = args.first
+            repo = opts[:repo]
+            druid = opts[:druid]
```

Comment:
This should probably be `opts.fetch(:druid)` because it should raise an error if you don't pass it.

Reply:
Done.